resources/kprove.mak

```diff
@@ -68,61 +84,80 @@ SPEC_FILES:=$(patsubst %,$(SPECS_DIR)/$(SPEC_GROUP)/%-spec.k,$(SPEC_NAMES))
 PANDOC_TANGLE_SUBMODULE:=$(ROOT)/.build/pandoc-tangle
 TANGLER:=$(PANDOC_TANGLE_SUBMODULE)/tangle.lua
 LUA_PATH:=$(PANDOC_TANGLE_SUBMODULE)/?.lua;;
+export TANGLER
 export LUA_PATH
 
 #
 # Dependencies
 #
 
-.PHONY: all clean clean-deps deps split-proof-tests test
+.PHONY: all clean clean-deps clean-k clean-kevm clean-kevm-cache deps deps-tangle deps-k deps-kevm split-proof-tests test
 
 all: deps split-proof-tests
 
 clean:
 	rm -rf $(SPECS_DIR)
 
-clean-deps:
-	rm -rf $(SPECS_DIR) $(K_REPO_DIR) $(KEVM_REPO_DIR)
+clean-deps: clean clean-k clean-kevm
+clean-k:
+	rm -rf $(K_REPO_DIR)
+clean-kevm:
+	rm -rf $(KEVM_REPO_DIR)
+clean-kevm-cache:
+	rm -rf $(KEVM_REPO_DIR)/.build/defn/java/driver-kompiled/cache.bin
 
-deps: $(K_REPO_DIR) $(KEVM_REPO_DIR) $(TANGLER)
+deps: deps-tangle deps-k deps-kevm
+deps-tangle: $(PANDOC_TANGLE_SUBMODULE)/submodule.timestamp
+deps-k:      $(K_REPO_DIR)/mvn.timestamp
+deps-kevm:   $(KEVM_REPO_DIR)/make.timestamp
 
-$(K_REPO_DIR):
-	git clone $(K_REPO_URL) $(K_REPO_DIR)
+%/submodule.timestamp:
+	git submodule update --init --recursive -- $*
+	touch $@
+
+$(K_REPO_DIR)/mvn.timestamp: $(K_VERSION_FILE) | $(K_REPO_DIR)
 	cd $(K_REPO_DIR) \
+		&& git fetch \
 		&& git reset --hard $(K_VERSION) \
 		&& git submodule update --init --recursive --remote \
 		&& mvn package -DskipTests -Dllvm.backend.skip -Dhaskell.backend.skip
+	touch $@
 
-$(KEVM_REPO_DIR):
-	git clone $(KEVM_REPO_URL) $(KEVM_REPO_DIR)
+$(K_REPO_DIR):
+	git clone $(K_REPO_URL) $(K_REPO_DIR)
+
+$(KEVM_REPO_DIR)/make.timestamp: $(KEVM_VERSION_FILE) $(K_REPO_DIR)/mvn.timestamp | $(KEVM_REPO_DIR)
 	cd $(KEVM_REPO_DIR) \
+		&& git fetch \
 		&& git clean -fdx \
 		&& git reset --hard $(KEVM_VERSION) \
 		&& make tangle-deps \
 		&& make defn \
 		&& $(K_BIN)/kompile -v --debug --backend java -I .build/defn/java -d .build/defn/java --main-module ETHEREUM-SIMULATION --syntax-module ETHEREUM-SIMULATION .build/defn/java/driver.k
+	touch $@
 
-$(TANGLER):
-	git submodule update --init -- $(PANDOC_TANGLE_SUBMODULE)
+$(KEVM_REPO_DIR):
+	git clone $(KEVM_REPO_URL) $(KEVM_REPO_DIR)
 
 #
 # Specs
 #
 
-split-proof-tests: $(SPECS_DIR)/$(SPEC_GROUP) $(SPECS_DIR)/lemmas.k $(SPEC_FILES)
+split-proof-tests: $(SPECS_DIR)/$(SPEC_GROUP)/lemmas.timestamp $(dir $(SPECS_DIR)/$(SPEC_GROUP))/lemmas.k $(SPEC_FILES)
```

Comment:
`$(SPECS_DIR)/lemmas.k` is also needed, no?

Reply:
No, the line above includes this most common case. Otherwise jenkins would have failed.